tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/java/com/acme/test/AnnotatedListener.java

```diff
@@ -83,84 +85,66 @@ public void contextInitialized(ServletContextEvent sce)
     @Override
     public void contextDestroyed(ServletContextEvent sce)
     {
-        // System.err.println("contextDestroyed "+sce);
+        LOG.fine("contextDestroyed " + sce);
     }
 
     @Override
     public void attributeAdded(ServletContextAttributeEvent scab)
     {
-        // System.err.println("attributeAdded "+scab);
+        LOG.fine("attributeAdded " + scab);
     }
 
     @Override
     public void attributeRemoved(ServletContextAttributeEvent scab)
     {
-        // System.err.println("attributeRemoved "+scab);
+        LOG.fine("attributeRemoved " + scab);
     }
 
     @Override
     public void attributeReplaced(ServletContextAttributeEvent scab)
     {
-        // System.err.println("attributeReplaced "+scab);
+        LOG.fine("attributeReplaced " + scab);
     }
 
     @Override
     public void requestDestroyed(ServletRequestEvent sre)
     {
-        // System.err.println("requestDestroyed "+sre);
+        LOG.fine("requestDestroyed " + sre);
     }
 
     @Override
     public void requestInitialized(ServletRequestEvent sre)
     {
-        // System.err.println("requestInitialized "+sre);
+        LOG.fine("requestInitialized " + sre);
     }
 
     @Override
     public void attributeAdded(ServletRequestAttributeEvent srae)
     {
-        // System.err.println("attributeAdded "+srae);
+        LOG.fine("attributeAdded " + srae);
     }
 
     @Override
     public void attributeRemoved(ServletRequestAttributeEvent srae)
     {
-        // System.err.println("attributeRemoved "+srae);
+        LOG.fine("attributeRemoved " + srae);
     }
 
     @Override
     public void attributeReplaced(ServletRequestAttributeEvent srae)
     {
-        // System.err.println("attributeReplaced "+srae);
+        LOG.fine("attributeReplaced " + srae);
     }
 
     @Override
     public void sessionCreated(HttpSessionEvent se)
     {
-        // System.err.println("sessionCreated "+se);
+        LOG.fine("sessionCreated " + se);
     }
 
     @Override
     public void sessionDestroyed(HttpSessionEvent se)
     {
-        // System.err.println("sessionDestroyed "+se);
-    }
-
-    public void requestCompleted(ServletRequestEvent rre)
-    {
-        // TODO Auto-generated method stub
-
-    }
-
-    public void requestResumed(ServletRequestEvent rre)
-    {
-        // TODO Auto-generated method stub
-
-    }
-
-    public void requestSuspended(ServletRequestEvent rre)
-    {
-        // TODO Auto-generated method stub
-
```

Comment:
Rather than delete these, you should also add logging statements for them just like the others.

Reply:
How are these called?
It's not part of a listener that the test class implements.